api/DeviceAPI.cs

```diff
@@ -34,18 +35,41 @@ public List<string> enumerateDevices(UInt64 house_id)
 	{
 		//TODO: Verify the input parameters are sufficient
 		//TODO: Implement this function
-		if (house_id == 0)
-	        {
-	            return null;
-	        }
-	        //get device list
-	        var devicelist = new List<string>();
-	        var devices = new List<Device>();
-	        var dlist =
-	            (from device in devices
-	             where device.ID.HouseID == house_id
-	             select device);
-	        return devicelist;
+        if (house_id == 0)
```

Comment:
Likely because for Room ID's, 0 *is* invalid. But for the other two parts of the ID, it isn't a magic number.